### PR TITLE
Create the plane primitive

### PR DESCRIPTION
**Pull request by Njord201**

### CODE REVIEW

File: include/Primitives/Plane.hpp [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+            /**
+             * @brief return the hit point of the plane
+             *
+             * @param ray ray to check vector3D
+             * @return Point3D
+             */
+            Math::Point3D hitPoint(const Raytracer::Ray& ray) const override;
+
+        private:
+            Axis _axis;
```
Comment (mathieurobert1) [inline include/Primitives/Plane.hpp:51]: The axis should not be a Vector3D ?

Comment (Njord201) [reply include/Primitives/Plane.hpp:51]: No it's the axis of the plane

**Review by mathieurobert1 — DISMISSED** [review]

File: include/Primitives/Plane.hpp [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+             * @brief Construct a new Plane object
+             *
+             */
+            Plane();
+
+            /**
+             * @brief Construct a new Plane object
+             *
+             */
+            Plane(Primitive::Axis axis, double position);
```
Comment (AxelF44) [inline include/Primitives/Plane.hpp:34]: Hello Mister Tranchet,

Can you please add comment about the variable that you use in your documentation

File: src/Plugins/Primitives/Plane/Plane.cpp [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    }
+    _axis = axis;
+}
+
+Math::Point3D Primitive::Plane::hitPoint(const Raytracer::Ray &ray) const
+{
+    Math::Point3D rayOrigin = ray.origin();
+    Math::Vector3D rayDirection = ray.direction();
+
+
```
Comment (AxelF44) [inline src/Plugins/Primitives/Plane/Plane.cpp:39]: Hello Mister Tranchet,

Why there is two space ?

**Review by AxelF44 — CHANGES_REQUESTED** [review]
Hello Mister Tranchet,

This is a very good and huge job. I Congrats you man !!!

File: src/Plugins/Primitives/Plane/Plane.cpp (hunk 3 above)
Comment (Njord201) [reply src/Plugins/Primitives/Plane/Plane.cpp:39]: Hello Mister Fradet,

That's a very good question, I remove one just after.

**Review by mathieurobert1 — APPROVED** [review]

**Review by AxelF44 — APPROVED** [review]

**Pull request merged**